tck/metrics/src/main/java/org/eclipse/microprofile/telemetry/metrics/tck/jvm/MetricsReader.java

```diff
@@ -21,36 +21,67 @@
  **********************************************************************/
 package org.eclipse.microprofile.telemetry.metrics.tck.jvm;
 
-import java.io.BufferedReader;
-import java.io.FileReader;
-import java.io.IOException;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.commons.io.input.Tailer;
+import org.apache.commons.io.input.TailerListenerAdapter;
+import org.awaitility.Awaitility;
+import org.testng.Assert;
 
 public class MetricsReader {
 
     private static final String logFilePath = System.getProperty("mptelemetry.tck.log.file.path");
 
-    public static boolean checkMessage(String metricName, String metricDescription, String metricUnit,
-            String metricType)
-            throws IOException {
-        try {
-            Thread.sleep(5000);
-            try {
-                BufferedReader reader = new BufferedReader(new FileReader(logFilePath));
-                String line;
-                while ((line = reader.readLine()) != null) {
-                    if (line.contains(
-                            "name=" + metricName + ", description=" + metricDescription + ", unit=" + metricUnit
-                                    + ", type=" + metricType)) {
-
-                        return true;
-                    }
-                }
-                return false;
-            } catch (IOException e) {
-                return false;
+    public static void checkMessage(String metricName, String metricDescription, String metricUnit,
+            String metricType) {
+
+        ExecutorService es = Executors.newFixedThreadPool(1);
+
+        LogFileTailerAdaptor logFileTailerAdaptor =
+                new LogFileTailerAdaptor(metricName, metricDescription, metricUnit, metricType);
+
+        Tailer tailer = Tailer.builder()
+                .setStartThread(true)
+                .setPath(logFilePath)
+                .setExecutorService(es)
+                .setReOpen(false)
+                .setTailerListener(logFileTailerAdaptor)
+                .setTailFromEnd(false)
+                .get();
+
+        try (tailer) {
+            Awaitility.await().pollDelay(3, SECONDS).atMost(15, SECONDS)
+                    .untilAsserted(() -> Assert.assertTrue(logFileTailerAdaptor.foundMetric()));
+        }
```

Comment:
This is nice. Does closing it here mean that you can remove `tailer.close()` from the `handle` method?

Reply:
It does. I deliberately left it in after checking to see if there's any risk of calling close twice (`close` just sets a boolean to false) because `handle` will close the connection faster than waiting for Awaitility to check the assert.